### PR TITLE
BaSP-TG-86: Add/edit tasks

### PR DESCRIPTION
**Pull request by GuidoCerioni**

### CODE REVIEW

File: src/components/Task/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          <SelectDropdown
+            name="Projects"
+            value={projectId}
+            onChange={(e) => {
+              setProjectId(e.target.value);
+            }}
+            options={projects.map((project) => ({
+              label: project.name,
+              value: project._id
+            }))}
```
Comment (franco-marini) [inline src/components/Task/index.js:130]: It is not bad to have it like this, but with each render of the app, the options would run this function each time. I recommend you to use this with a useEffect and a useState

File: src/components/Task/index.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  const [projectId, setProjectId] = useState('');
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [requestType, setRequestType] = useState('POST');
+  const [modalTitle, setModalTitle] = useState('');
+  const [modalText, setModalText] = useState('');
+
+  useEffect(() => {
+    dispatch(getProjects());
+  }, []);
```
Comment (franco-marini) [inline src/components/Task/index.js:37]: Here what you can do is the same as we have on the list entity page, and also format and save the list on the state as for example `projectsOptions`. What do you think?

File: src/components/Task/index.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      setProjectId(task.projectId._id);
+      setRequestType('PUT');
+    }
+  }, [error]);
+
+  const formatDate = (date) => {
+    var d = new Date(date),
+      month = '' + (d.getMonth() + 1),
+      day = '' + d.getDate(),
+      year = d.getFullYear();
```
Comment (franco-marini) [inline src/components/Task/index.js:53]: Please do not use `var`, use `let` and `const` instead

**Review by franco-marini — CHANGES_REQUESTED** [review]
Eggcellent! Just some little comments to polish the code and be more performance

**Review by franco-marini — APPROVED** [review]
Nice work!

**Review by Facund0Fernandez — APPROVED** [review]

**Review by AlejandroSantini — APPROVED** [review]
Nice job!

**Review by vadalamati — APPROVED** [review]
Good job!

**Pull request merged**